src/app/forms/validators/custom-async-validators.ts

```diff
@@ -114,7 +117,52 @@ export class CustomAsyncValidators {
 
           return null;
         })
+   
+        );
+  }
+
+  static hideIfEqualsWithCondition(store: Store<State>, sibling: string, value: string, conditions: Condition | Condition[]): AsyncValidatorFn {
+    return (control: AbstractControl): Observable<ValidationErrors | null> =>
+      store.pipe(
+        take(1),
+        select(testResultInEdit),
+        map(testResult => {
+          if (!testResult || !control?.parent) { return null; }
+
+          const siblingControl = control.parent.get(sibling) as CustomFormControl;
+
+          const conditionsPassed = CustomAsyncValidators.checkConditions(testResult, conditions);
+
+          siblingControl.meta.hide = conditionsPassed && (Array.isArray(value) ? value.includes(control.value) : control.value === value);
+
+          return null;
+        })
       );
   }
- 
+
+  static checkConditions(testResult: TestResultModel, conditions: Condition | Condition[] ){
+    if (!Array.isArray(conditions)) {
+      return CustomAsyncValidators.checkCondition(testResult, conditions);
+    }
+
+    for (const condition of conditions) {
+      if(!CustomAsyncValidators.checkCondition(testResult, condition)) {
+        return false;
+      }
+    }
+    
+    return true;
```

Comment:
I think this is the same logic?
```suggestion
    return conditions.every(condition => CustomAsyncValidators.checkCondition(testResult,condition))

Reply:
Agreed

Reply:
Thanks Guillaume that looks good. I tested it locally and all good so I will commit that.